### PR TITLE
feat: adding relevant files for allow list

### PR DESCRIPTION
**Pull request by Nilay27**
* Adding relevant files for allow list contract.

### CODE REVIEW

File: apps/smart-contracts/token/contracts/Allowlist.sol [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+contract Allowlist is IAllowlist, SafeOwnable {
+  constructor(address _nominatedOwner) {}
+
+  function setDestinations(address[] calldata _destinations, bool[] calldata _allowed)
+    external
+    override
+    onlyOwner
+  {}
+
+  function resetDestinations(address[] calldata _allowedDestinations) external override onlyOwner {}
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/token/contracts/Allowlist.sol:16]:
```suggestion
  function resetDestinations(address[] calldata _destinationsToAllow) external override onlyOwner {}
```

File: apps/smart-contracts/token/contracts/Allowlist.sol [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  function setSources(address[] calldata _sources, bool[] calldata _allowed)
+    external
+    override
+    onlyOwner
+  {}
+
+  function resetSources(address[] calldata _allowedSources) external override onlyOwner {}
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/token/contracts/Allowlist.sol:24]:
```suggestion
  function resetSources(address[] calldata _sourcesToAllow) external override onlyOwner {}
```

**Review by ramenforbreakfast — APPROVED** [review]
LGTM, besides nits around `reset...()` parameter naming.

**Review by XavierEkkel — APPROVED** [review]
LGTM

**Comment by ramenforbreakfast** [comment]
closing due to decision to switch over to Accesslist

**Pull request closed without merging**